server/controllers/finance/purchase.js

```diff
@@ -3,152 +3,213 @@ var db = require('./../../lib/db');
-    // shipped
-    case 'paid':
-      sql += ' AND paid = 1;';
-      break;
-
-    // the purchased items have been shipped, but have not
-    // arrived in the warehouse yet
-    case 'shipped':
-      sql += ' AND closed = 0 AND confirmed = 1;';
-      break;
-
-    // the purchase order has been fulfilled with stock in the
-    // warehouse.
-    case 'delivered':
-      sql += ' AND closed = 1 AND confirmed = 1;';
-      break;
-
-    default:
-      sql += ';';
-      break;
+  var purchase = req.body;
+  var transaction;  
+  var purchaseOrder = purchase.purchase_order;
+  var purchaseItem =  purchase.purchase_item;
+
+  // Reject invalid parameters
+  if (!purchaseItem || !purchaseItem) { 
+    res.status(400).json({
+      code : 'ERROR.ERR_MISSING_INFO', 
+      reason : 'A valid sale details and sale items must be provided under the attributes `sale` and `saleItems`'
+    });
+    return;
   }
+  
 
-  db.exec(sql)
-  .then(function (rows) {
-    res.status(200).send(rows);
-  })
-  .catch(next)
-  .done();
-};
 
-/*
- * Utility Methods
-*/
-// TODO Server side sale and purchase order are very similar, they should
-// probably be combined
+  var sqlPurchase = 'INSERT INTO purchase SET ?';
 
-// TODO employee responsible for purchase, credited money from purchase request
-//      must be able to select them at purchase order
+  var sqlPurchaseItem = 'INSERT INTO purchase_item (uuid, inventory_uuid, quantity, unit_price, ' +
+    'total, purchase_uuid) VALUES ?'; 
 
-// TODO create purchase order, expense cash, confirm recieving (work with overflow money)
+  var dataPurchaseItem = linkPurchaseItems(purchase.purchase_item, purchaseOrder.uuid);
 
-// TODO
-//  - Purchase price and sale price
-//  - On purchase order no money is moved
-//  - Move money from cash balance account to asset account (on authorisation of purchase)
-//  - On sale move money from asset account to expense account,
-//    credit sale account
+  transaction = db.transaction();
 
-function values(obj) {
-  // get an array of an object's values
-  // cannot wait for es6's Object.values(o);
-  return Object.keys(obj).map(function (k) { return obj[k]; });
-}
+  transaction
+    .addQuery(sqlPurchase, [purchaseOrder])
+    .addQuery(sqlPurchaseItem,[linkPurchaseItems(purchaseItem, purchaseOrder.uuid)]);
 
-function executePurchase(userId, data) {
-  // Writes data to the primary_cash table
-  // then the primary_cash_items table and
-  // finally the posting_journal
-  var primaryCashId = uuid();
+  transaction.execute()
+    .then(function (results) {
+      var confirmation =  { 
+        uuid : purchaseOrder.uuid,
+        results : results
+      };
 
-  return writeCash(primaryCashId, userId, data.details)
-    .then(function () {
-      return writeCashItems(primaryCashId, data.transaction);
-    })
-    .then(function () {
-      return postToJournal(primaryCashId, userId);
+      res.status(201).json(confirmation);
```

Comment:
This should only send back the uuid.  Something like this:
```js
res.status(201).json({ uuid : purchaseOrder.uuid });
```
The `results` object doesn't have anything the client will be interested in.  No need to send it back.